### PR TITLE
Merge branch-edit-buyer-command to master

### PR DESCRIPTION
**Pull request by lokidoki102**
The code for edit buyer command is now fully functional and ready to be deployed to master.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/EditBuyerCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditBuyerCommand.MESSAGE_USAGE), pe);
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_HOUSING_TYPE,
+                PREFIX_BUDGET);
+
+        EditBuyerDescriptor editBuyerDescriptor = new EditBuyerDescriptor();
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editBuyerDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
```
Comment (redcolorbicycle) [inline src/main/java/seedu/address/logic/parser/EditBuyerCommandParser.java:53]: Any way to compress the if statements or abstract them away? If not looks okay

File: src/test/java/seedu/address/testutil/EditBuyerDescriptorBuilder.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     * that we are building.
+     */
+    public EditBuyerDescriptorBuilder withTags(String... tags) {
+        Set<Tag> tagSet = Stream.of(tags).map(Tag::new).collect(Collectors.toSet());
+        this.descriptor.setTags(tagSet);
+        return this;
+    }
+
+    public EditBuyerDescriptor build() {
+        return this.descriptor;
```
Comment (redcolorbicycle) [inline src/test/java/seedu/address/testutil/EditBuyerDescriptorBuilder.java:87]: LGTM

**Review by redcolorbicycle — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/EditBuyerCommandParser.java (hunk 1 above)
Comment (lokidoki102) [reply src/main/java/seedu/address/logic/parser/EditBuyerCommandParser.java:53]: Yeahh I guess the chaining is too long.

Would it be better if I split it this way?
```
        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
        
        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
            editBuyerDescriptor.setName(name);
        }
```

**Review by felixchanyy — APPROVED** [review]
LGTM

**Pull request merged**